component/storage/mongodb/store_test.go

```diff
@@ -0,0 +1,303 @@
+/*
+Copyright Scoir Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package mongodb_test
+
+import (
+	"context"
+	"log"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/hyperledger/aries-framework-go/spi/storage"
+	commontest "github.com/hyperledger/aries-framework-go/test/component/storage"
+	dctest "github.com/ory/dockertest/v3"
+	dc "github.com/ory/dockertest/v3/docker"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/hyperledger/aries-framework-go-ext/component/storage/mongodb"
+)
+
+const (
+	mongoDBConnString    = "mongodb://localhost:27017"
+	dockerMongoDBImage   = "mongo"
+	dockerMongoDBTagV400 = "4.0.0"
+	dockerMongoDBTagV428 = "4.2.8"
+	dockerMongoDBTagV500 = "5.0.0"
+)
+
+// This should function the same as the default logger in the mongodb package.
+// This is here just to increase code coverage by allowing us to exercise the WithLogger option.
+type testLogger struct {
+	logger *log.Logger
+}
+
+func (d *testLogger) Infof(msg string, args ...interface{}) {
+	d.logger.Printf(msg, args...)
+}
+
+func TestMongoDB_V4_0_0(t *testing.T) {
+	startContainerAndDoAllTests(t, dockerMongoDBTagV400)
+}
+
+func TestMongoDB_V4_2_8(t *testing.T) {
+	startContainerAndDoAllTests(t, dockerMongoDBTagV428)
+}
+
+func TestMongoDB_V5_0_0(t *testing.T) {
```

Comment:
I also ran all the same tests against DocumentDB on my system locally, but it can't be committed since it required a SSH tunnel and a private DocumentDB instance (and there are no Docker images for it).